internal/provider/kubernetes/controller.go

```diff
@@ -368,37 +252,139 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, _ reconcile.Reques
+	for backendRef := range resourceMappings.allAssociatedBackendRefs {
+		backendRefKind := gatewayapi.KindDerefOr(backendRef.Kind, gatewayapi.KindService)
+		r.log.Info("processing Backend", "kind", backendRefKind, "namespace", string(*backendRef.Namespace),
+			"name", string(backendRef.Name))
+
+		var endpointSliceLabelKey string
+		switch backendRefKind {
+		case gatewayapi.KindService:
+			service := new(corev1.Service)
+			err := r.client.Get(ctx, types.NamespacedName{Namespace: string(*backendRef.Namespace), Name: string(backendRef.Name)}, service)
+			if err != nil {
+				r.log.Error(err, "failed to get Service", "namespace", string(*backendRef.Namespace),
+					"name", string(backendRef.Name))
+			} else {
+				resourceMappings.allAssociatedNamespaces[service.Namespace] = struct{}{}
+				gwcResource.Services = append(gwcResource.Services, service)
+				r.log.Info("added Service to resource tree", "namespace", string(*backendRef.Namespace),
+					"name", string(backendRef.Name))
+			}
+			endpointSliceLabelKey = discoveryv1.LabelServiceName
+
+		case gatewayapi.KindServiceImport:
+			serviceImport := new(mcsapi.ServiceImport)
+			err := r.client.Get(ctx, types.NamespacedName{Namespace: string(*backendRef.Namespace), Name: string(backendRef.Name)}, serviceImport)
+			if err != nil {
+				r.log.Error(err, "failed to get ServiceImport", "namespace", string(*backendRef.Namespace),
+					"name", string(backendRef.Name))
+			} else {
+				resourceMappings.allAssociatedNamespaces[serviceImport.Namespace] = struct{}{}
+				gwcResource.ServiceImports = append(gwcResource.ServiceImports, serviceImport)
+				r.log.Info("added ServiceImport to resource tree", "namespace", string(*backendRef.Namespace),
+					"name", string(backendRef.Name))
+			}
+			endpointSliceLabelKey = mcsapi.LabelServiceName
+		}
+
+		// Retrieve the EndpointSlices associated with the service
+		endpointSliceList := new(discoveryv1.EndpointSliceList)
+		opts := []client.ListOption{
+			client.MatchingLabels(map[string]string{
+				endpointSliceLabelKey: string(backendRef.Name),
+			}),
+			client.InNamespace(string(*backendRef.Namespace)),
+		}
+		if err := r.client.List(ctx, endpointSliceList, opts...); err != nil {
+			r.log.Error(err, "failed to get EndpointSlices", "namespace", string(*backendRef.Namespace),
+				backendRefKind, string(backendRef.Name))
+		} else {
+			for _, endpointSlice := range endpointSliceList.Items {
+				endpointSlice := endpointSlice
+				r.log.Info("added EndpointSlice to resource tree", "namespace", endpointSlice.Namespace,
+					"name", endpointSlice.Name)
+				gwcResource.EndpointSlices = append(gwcResource.EndpointSlices, &endpointSlice)
+			}
+		}
+	}
+}
+
+// processSecurityPolicyObjectRefs adds the referenced resources in SecurityPolicies
 // to the resourceTree
-func (r *gatewayAPIReconciler) processSecurityPolicySecretRefs(
+// - Secrets for OIDC and BasicAuth
```

Comment:
you probably need to also make sure that the controller reconciles if the ext auth backend ref changes :)
so the indexer needs to be added or updated similar to https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/provider/kubernetes/indexers.go#L380

Reply:
@arkodg  It's already there https://github.com/envoyproxy/gateway/blob/ef1934f1461157afc890cddac863418c3e90b4f0/internal/provider/kubernetes/indexers.go#L406